src/main/java/org/disq_bio/disq/HtsjdkVariantsRdd.java

```diff
@@ -0,0 +1,33 @@
+package org.disq_bio.disq;
+
+import htsjdk.variant.variantcontext.VariantContext;
+import htsjdk.variant.vcf.VCFHeader;
+import org.apache.spark.api.java.JavaRDD;
+
+/**
+ * A {@link HtsjdkVariantsRdd} is the distributed equivalent of a htsjdk {@link
+ * htsjdk.variant.vcf.VCFFileReader}. It represents a VCF file stored in a distributed filesystem,
+ * and encapsulates a Spark RDD containing the variant records in it.
+ *
+ * <p>Use a {@link HtsjdkVariantsRddStorage} to read and write {@link HtsjdkVariantsRdd}s.
+ *
+ * @see HtsjdkVariantsRddStorage
+ */
+public class HtsjdkVariantsRdd {
```

Comment:
This could be a common interface for reads/variants and any other kind of records (it will be nice to have FASTQ support eventually and other tribble features). What's about using a `RecordsRdd<Metadata, Record>`? For reads, it will be `RecordsRdd<SAMFileHeader, SAMRecord>` and for variants `RecordsRdd<VCFHeader, VariantContext>` - that is more extensible for other formats (and for moving to a different library, or htsjdk-next, in the future).

Reply:
Yes, that's a good idea.

Reply:
This probably needs more discussion. Should `HtsjdkReadsRdd` still exist and implement the interface? Should its only methods be `getMetadata()` and `getRecords()`, or should it also have `getHeader()` and `getReads()`? I'd be open to evolving this later - I think it's OK for the API to be unstable for a while.

Reply:
I think that by now the best option is to keep `HtsjdkReadsRdd` extending the interface and with the more explicit methods (getHeader/getReads) and then decide if it is really required to have the implementation. This is in line with some discussion that we had in htsjdk-next-beta, where we are planning to add a high-level abstraction for every record reader/iterator with common methods. Maybe at some point `disq` can extend directly that interfaces to make, when possible, processing on spark/hadoop more similar to processing on a local machine.

Reply:
We are discussing this design of readers in slack
@lbergelson, you were the one proposing a `RecordIterator<Record, Metadata>` and `RecordReader<Record, Metadata, QueryKey>`
@tomwhite - if you want, I can ask the rest of maintainers to add you to the slack chat (I think that it will be fine)